server/src/tree-sitter/analyzer.ts

```diff
@@ -707,62 +711,92 @@ export default class Analyzer {
       return undefined
     }
 
-    const validTexts = lines.slice(index).join('\n')
-    const scanResultdocumentUri = uriString + '.scanned'
-    const scanResultDocument = TextDocument.create(
-      scanResultdocumentUri,
-      'bitbake',
-      0,
-      validTexts
-    )
+    const scanResultDoc = lines.slice(index).join('\n')
+    const scanResultDocUri = originalDocUri + '.scanned'
+    const scanResultDocTree = this.parser.parse(scanResultDoc)
 
-    this.analyze({ document: scanResultDocument, uri: scanResultdocumentUri })
+    const [scanResultDocGlobalDeclarations] = getGlobalDeclarationsAndComments({ tree: scanResultDocTree, uri: scanResultDocUri, getFinalValue: true })
+    const scanResultDocSymbols = this.getAllSymbolsFromGlobalDeclarations(scanResultDocGlobalDeclarations)
+    const { globalDeclarations: analyzedOriginalDocGlobalDeclarations } = analyzedOriginalDoc
 
-    const scanResultDocumentSymbols = this.getGlobalDeclarationSymbols(scanResultdocumentUri)
-    if (scanResultDocumentSymbols.length > 0) {
-      logger.debug(`symbol length for scan result document: ${scanResultDocumentSymbols.length}`)
-    }
+    const processedSymbolInformation: RequestResult['ProcessRecipeScanResults'] = {}
 
-    // Symbol information on the file that was requested to be scanned
-    const originalSymbols = this.getGlobalDeclarationSymbols(uriString)
+    // Replace final values of the symbols in the original document with the ones from the scan results
+    Object.values(analyzedOriginalDocGlobalDeclarations).forEach((symbolArray) => {
+      symbolArray.forEach((symbol) => {
+        if (symbol.kind === SymbolKind.Variable) {
+          const sameSymbolInScanResultDocument = scanResultDocSymbols.find((scanResultDocumentSymbol) => {
+            return scanResultDocumentSymbol.name === symbol.name &&
+            scanResultDocumentSymbol.overrides?.length === symbol.overrides?.length &&
+            scanResultDocumentSymbol.overrides?.every((override) => symbol.overrides?.includes(override)) // Question: should we care about the order of the overrides?
+          })
+          if (sameSymbolInScanResultDocument !== undefined) {
+            symbol.finalValue = sameSymbolInScanResultDocument.finalValue
 
-    if (originalSymbols.length > 0) {
-      logger.debug(`symbol length for original document: ${originalSymbols.length}`)
-    }
+            if (processedSymbolInformation[symbol.name] === undefined) {
+              processedSymbolInformation[symbol.name] = []
+            }
 
-    // TODO: Further parsing of the scan results using both documents
-    return undefined
+            (processedSymbolInformation[symbol.name] as BitbakeSymbolInformation[]).push(symbol)
+          }
+        }
+      })
+    })
+
+    /**
+     TODO: Achieve the following goals using the scan results
+       [check] 1. Get the final values for variables and show them in the hover
```

Comment:
I didn't get to see any hovers with final values. I tried:
 - running the command for the busybox recipe
 - hovering on variable assignments
 - hovering on variable references

Reply:
A presentation idea: could you instead change the value format to be presented as a string (which it is). For instance, when I use the hover from the JS debugger, you can see 'busybox' here is brown.
![image](https://github.com/yoctoproject/vscode-bitbake/assets/19970889/65ff3cf5-4520-4a3f-a124-160024206aa7)

Reply:
Ok, I got the result to show using the contextual menu command, which works nicely. However it doesn't seem to provide the results if I start the command through Ctrl+Shift+P.

Reply:
Done. Check b1b1fd2fdca0bfda2786607f8bb4881cc6aa0320

Reply:
Check fd2276db9d1737d269333eb3ad295d32f7113636 for applying the string color to the text

Reply:
yes! Good job